outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/service/impl/TargetFileServiceImpl.java

```diff
@@ -144,6 +150,27 @@ private File createTargetFileDirectory() {
     }
 
 
+    private void writeSubscriptionRow(Subscription subscription, OutputStreamWriter writer) throws IOException {
+        //subscription id
+        writer.write(subscription.getSubscriptionId());
+        writer.write(",");
+
+        //phone number
+        writer.write(subscription.getSubscriber().getCallingNumber().toString());
+        writer.write(",");
+
+        //language
+        Subscriber subscriber = subscription.getSubscriber();
+        //todo: don't understand why subscriber.getLanguage() doesn't work here...
+        Language language = (Language) subscriberDataService.getDetachedField(subscriber, "language");
```

Comment:
I put a ticket in.  There is something that causes a 1:1 to not get detached if there is a 1:M relationship as well.

https://applab.atlassian.net/browse/MOTECH-1678